### PR TITLE
Feature/cce 639 implemetn scripts

### PR DESCRIPTION
**Pull request by like-me1**
Sorry, I messed up a feature branch with feature/CCE-639-implement-scripts, please review this one. Conflicts are resolved and it's up to date with the current v2.0 branch. Shouldn't cause any issues.
This PR is a first run for the review. It's a massive task and I believe there will be some parts forgotten. But we could clean this PR up and move to the next ticket everything that is left. The main goal of this task was to implement skeletor, no details of animation, interactions, responsiveness etc. I didn't use Bootstrap so far and decreased the number of used classes in comparison to what we have online.

### CODE REVIEW

File: docs/styleguide/includes/app_header.html [hunk 1]
```diff
@@ -29,12 +30,21 @@
...
+            <img src="svg/logo-hybris.svg" class="brand-logo" alt="SAP Hybris">
+        </a>
+        <!-- mobile navigation -->
+        <a href="#" class="primary-nav--toggle">
+            <span class="primary-nav-triangle"></span>
         </a>
     </div>
 
     <div class="docs-primary-nav">
+        <a id="menu-toggle">
```
Comment (xak) [inline docs/styleguide/includes/app_header.html]: Same thing here ...
```
<button id="menu-toggle" 
   data-js="menu-toggle"
   aria-label="Toggle Menu" 
   aria-controls="yaas-side-nav"
   aria-expanded="false">
...
</button>
```

File: docs/styleguide/includes/app_header.html [hunk 2]
```diff
@@ -3,22 +3,23 @@
...
         {% set signav = app.nav.yaas %}
         {% set home = signav[0] %}
-        <a href="{{ home.url }}" class="yaas-signature-nav__index {{ home.id }}--">
-            <img src="svg/logo-yaas.svg" height="30" alt="YaaS">YaaS
+        <a href="{{ home.url }}" class="yaas-signature-nav__index domain">
+            <img src="svg/logo-yaas.svg" class="domain-logo" alt="YaaS">
+            <span class="logo-text">YAAS</span>
         </a>
 
         <ul class="yaas-signature-nav__list" role="navigation">
```
Comment (xak) [inline docs/styleguide/includes/app_header.html]: ARIA here ...
```
<ul class="yaas-signature-nav__list" role="navigation" aria-hidden="true">
...
</ul>
```
* `aria-hidden` will replace the `open` class

File: docs/styleguide/templates/master.html [hunk 3]
```diff
@@ -43,13 +43,37 @@
 {% block header %}
   {% include "app_header.html" %}
 {% endblock %}
-    <main class="docs-app__main" role="main">
-
+    <main class="docs-app__main container-fluid" role="main">
+        <div class="mobile-left-side-navigation">
```
Comment (xak) [inline docs/styleguide/templates/master.html]: ARIA here ...
```
<div class="mobile-left-side-navigation" aria-hidden="true">
...
</div>
```
* `aria-hidden` will replace the `open` class

Comment (xak) [inline docs/styleguide/templates/master.html]: Can this go before `main`? The `main` container should be reserved for page-specific content, all app-level stuff should be outside of it.

And please pull this into an include to keep the master template at a very high-level.

File: docs/styleguide/templates/master.html [hunk 4]
```diff
@@ -43,13 +43,37 @@
...
+            <aside>
```
Comment (xak) [inline docs/styleguide/templates/master.html]: Make this a `nav` element.

File: docs/styleguide/scripts/main.js [hunk 5]
```diff
@@ -1,5 +1,13 @@
-/**
- * Created by d063436 on 5/21/17.
- */
-console.log('Just for js test purpose');
 
+$(document).ready(function(){
+    var primaryNavigationDropdown = $('.yaas-signature-nav__list');
+    $('.primary-nav--toggle').on('click', function(){
```
Comment (xak) [inline docs/styleguide/scripts/main.js]: Here you can pick up the `data-js="menu-toggle"` attribute to initialize the JS ...
```
$('[data-js="menu-toggle"]')
```
And then operate on the ARIA attributes ...
* You can find the target list from the `aria-controls` attribute (so you don't have to hardcode that in the JS)
* On click, change  `aria-expanded` on the trigger element — `$(this)` — to true
* Then on the target element, change `aria-hidden` to false
Of course, reverse when closing.

This way you aren't hardcoding a single class and you get accessibility for free.

File: docs/styleguide/scripts/main.js [hunk 6]
```diff
@@ -1,5 +1,13 @@
...
+        $(this).toggleClass('active');
+        $(primaryNavigationDropdown).toggleClass('open');
+    });
+
+    $('#menu-toggle').on('click', function(){
```
Comment (xak) [inline docs/styleguide/scripts/main.js]: You may not need this at all, it looks like both of these share the exact same behavior. In fact, this same script could be used for lots of toggle behaviors.

File: docs/styleguide/styles/_primary-nav.scss [hunk 7]
```diff
@@ -58,7 +174,146 @@
...
+            }
+        }
+    }
+    .docs-primary-nav__list{
+        display: none!important;
+    }
+    .primary-nav--toggle{
+        visibility: visible !important;
+        opacity: 1 !important;
+        &.active{
```
Comment (xak) [inline docs/styleguide/styles/_primary-nav.scss]: Change this to `&[aria-expanded="true"]`.

File: docs/styleguide/styles/_primary-nav.scss [hunk 8]
```diff
@@ -58,7 +174,146 @@
...
+@media (max-width: 740px) and (min-width: 0){
+    .yaas-signature-nav__list{
+        border-top: 1px solid #808fa1;
+        display: none!important;
+        width: 100%;
+        position: absolute;
+        top: $primary-nav--height;
+        z-index: 1001;
+        border-left: 0 none;
+        &.open{
```
Comment (xak) [inline docs/styleguide/styles/_primary-nav.scss]: Change this to `&[aria-hidden="false"]`.

File: docs/styleguide/styles/_primary-nav.scss [hunk 9]
```diff
@@ -58,7 +174,146 @@
...
+        background: none repeat scroll 0 0 #233243;
+        height: 2px;
+        transform: none;
+        width: 23px;
+        border-radius: 1px;
+        display: block;
+        margin-top: 4px;
+        transition: all 0.5s ease;
+    }
+    &.open{
```
Comment (xak) [inline docs/styleguide/styles/_primary-nav.scss]: Change this to `&[aria-expanded="true"]`.

File: docs/styleguide/styles/_primary-nav.scss [hunk 10]
```diff
@@ -58,7 +174,146 @@
...
+                padding: 10px 15px 10px 31px;
+                text-decoration: none;
+                font-size: 15px;
+                &:hover, &[aria-selected=true] {
+                    color: $default-blue;
+                }
+            }
+        }
+    }
+    &.open{
```
Comment (xak) [inline docs/styleguide/styles/_primary-nav.scss]: Change this to `&[aria-hidden="false"]`

File: src/styles/core/_settings.scss [hunk 11]
```diff
@@ -1,3 +1,4 @@
+@import "variables";
```
Comment (xak) [inline src/styles/core/_settings.scss]: # This file should not be changed at all, this is for the library, not the site.

File: src/styles/core/_variables.scss [hunk 12]
```diff
@@ -0,0 +1,41 @@
+//== DEFAULT COLOUR SETS
```
Comment (xak) [inline src/styles/core/_variables.scss]: # This file is not needed for the library.

File: docs/styleguide/templates/master.html [hunk 13]
```diff
@@ -61,7 +85,8 @@
 })();
     </script>
 {% endif %}
-<script src="js/techne-styleguide{% if env == 'production' %}.min{% endif %}.js"></script>
+    <script src="https://code.jquery.com/jquery-3.2.1.min.js"></script>
```
Comment (xak) [inline docs/styleguide/templates/master.html]: Remove the `https` protocol — `<script src="//code.jquery.com/jquery-3.2.1.min.js"></script>`

File: docs/styleguide/styles/_variables.scss [hunk 14]
```diff
@@ -0,0 +1,24 @@
+//all variable related to Web site techne.yaas.io and techne library overwrites should be located here
+
+//size
+$primary-nav--height: 50px;
+
+//fonts
+$font-family-dosis: 'Dosis', sans-serif;
+$font-size-base:          14px;
+$font-size--footer: ceil(($font-size-base * 1.07));
```
Comment (xak) [inline docs/styleguide/styles/_variables.scss]: Just put this in `footer.scss` if this variable will not be used anywhere else. And why not just make it `15px`? Being a global nav (across YaaS), it makes sense to hardcode these values.

**Comment by like-me1** [comment]
@xak I'm done with changes. Please review

**Comment by xak** [comment]
@like-me1 I'm going to merge this but there are some visual inconsistencies. When comparing with [YaaS > Tools](https://devportal.yaas.io/tools/) ...

- Logo SVG is misaligned (flush to the top)
- Logo `alt` should be "Techne" not "YaaS"
- "YAAS" should be white
- Dotted divider should be grayish color
- Top nav items shouldn't be 17px and normal weight
- Top nav hover states should be more like a tab (not filled fully to the top of the nav)
- Items should have more spacing around them
- SAP logo is misaligned
- Second-level nav items look a tad brighter and selected state shouldn't be bold
- At about 870px browser width the top nav breaks (becomes taller before the SAP logo is hidden)
- The responsive nav has some inconsistencies as well

Can you open an issue to cover the remaining tasks? You may have some other things on your list as well.

Also, the secondary nav background has disappeared ... 
screen shot 2017-06-16 at 11 30 35 am

**Pull request merged**